File: frontend/src/types/media.ts
export type MediaType = "movie" | "series";
export type WatchStatus = "watched" | "watching" | "planned" | "dropped";
export type ViewMode = "list" | "grid";

export interface MediaItem {
  id: number;
  title: string;
  media_type: MediaType;
  year: number | null;
  genres: string[];
  poster_url: string | null;
  rating: number | null;
  watch_status: WatchStatus | null;
  total_episodes: number | null;
  watched_episodes: number | null;
}

export interface MediaListResponse {
  items: MediaItem[];
  total: number;
}

export interface EpisodeDetail {
  number: number;
  title: string;
  air_date: string | null;
  still_url: string | null;
  watch_status: WatchStatus | null;
}

export interface SeasonDetail {
  number: number;
  poster_url: string | null;
  vote_average: number | null;
  release_date: string | null;
  total_episodes: number;
  watched_episodes: number;
  episodes: EpisodeDetail[];
}

export interface MediaDetailResponse {
  id: number;
  media_type: MediaType;
  title: string;
  year: number | null;
  poster_url: string | null;
  backdrop_path: string | null;
  overview: string | null;
  genres: string[];
  status: string | null;
  tmdb_rating_percent: number | null;
  watch_status: WatchStatus | null;
  tmdb_id: string | null;
  imdb_id: string | null;
  tvdb_id: string | null;
  seasons: SeasonDetail[];
}

File: backend/app/schemas/media.py
from typing import Literal

from pydantic import BaseModel


class MediaItem(BaseModel):
    id: int
    title: str
    media_type: Literal["movie", "series"]
    year: int | None = None
    genres: list[str] = []
    poster_url: str | None = None
    rating: float | None = None
    watch_status: Literal["watched", "watching", "planned", "dropped"] | None = None
    total_episodes: int | None = None
    watched_episodes: int | None = None


class MediaListResponse(BaseModel):
    items: list[MediaItem]
    total: int


class MediaDetailResponse(BaseModel):
    id: int
    media_type: Literal["movie", "series"]
    title: str
    year: int | None = None
    poster_url: str | None = None
    backdrop_path: str | None = None
    overview: str | None = None
    genres: list[str] = []
    status: str | None = None
    tmdb_rating_percent: int | None = None
    watch_status: Literal["watched", "watching", "planned", "dropped"] | None = None
    tmdb_id: str | None = None
    imdb_id: str | None = None
    tvdb_id: str | None = None
